File: packages/java/src/main/java/com/ailabtools/sdk/response/ErrorDetail.java
package com.ailabtools.sdk.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/** Structured error details returned by the AILabTools API. */
@JsonIgnoreProperties(ignoreUnknown = true)
public final class ErrorDetail {
    @JsonProperty("status_code")
    private Integer statusCode;
    private String code;
    @JsonProperty("code_message")
    private String codeMessage;
    private String message;

    public Integer getStatusCode() { return statusCode; }
    public String getCode() { return code; }
    public String getCodeMessage() { return codeMessage; }
    public String getMessage() { return message; }
}

File: packages/php/src/Response/ErrorDetail.php
<?php

declare(strict_types=1);

namespace AILabTools\Response;

final class ErrorDetail
{
    public function __construct(
        public readonly ?int $statusCode = null,
        public readonly ?string $code = null,
        public readonly ?string $codeMessage = null,
        public readonly ?string $message = null,
    ) {
    }

    /** @param array<string, mixed> $payload */
    public static function fromArray(array $payload): self
    {
        return new self(
            statusCode: isset($payload['status_code']) ? (int) $payload['status_code'] : null,
            code: isset($payload['code']) ? (string) $payload['code'] : null,
            codeMessage: isset($payload['code_message']) ? (string) $payload['code_message'] : null,
            message: isset($payload['message']) ? (string) $payload['message'] : null,
        );
    }
}
